### DISCUSSION TITLE
Setting MinSpeed for ReleaseSw

### DISCUSSION THREAD

**Discussion by halpb** [ponoor/step-series-support#4] [Q&A] [answered]
Hello-

Love the STEP400 and the documentation.  I have a NEMA17 stepper and an attached 10:1 gearbox.  I'd like to be able to increase the speed at which the 2nd part of the /homing operation (/releaseSw) moves the motor.  As the documentation states, "Move at minimum speed (5[step/s] by default)".  Due to my gearing, this is incredibly slow and I find it to be setting the home unreliably.  

I'd love to increase this to something higher, but I cannot see any mention of the command to do this?  I've tried /setMinSpeed but that does not appear to be a valid command.  

Any help is appreciated!

-Brett

**Comment by kanta** [comment, marked as answer]
Hello Brett,

Thank you for choosing the STEP400!
As you mentioned, `/releaseSw` command produces a motion at minimum speed. There is no OSC command to set it and this problem never become apparent in my environment. so it was very helpful to have it pointed out.

The best solution might be update the firmware for adding `/setMinSpeed` and I will do that. I will try to finish it soon but you will have to download/compile/write it to your board.

Another solution is to use following commands instead of `/homing`.

1. [`/enableHomeSwReport`](https://ponoor.com/en/docs/step-series/osc-command-reference/home-limit-sensors/#enablehomeswreport_intmotorid_boolenable) to enable notification when conditional changes of the HOME sensor. 
2. `/goUntil` to move the motor towards the home sensor.
3. After the process ends, you will get the response of [`/homeSw`](https://ponoor.com/en/docs/step-series/osc-command-reference/home-limit-sensors/#gethomesw_intmotorid). the swState will be 1.
4. `/run` to move the motor in the direction away from the sensor with the desired speed.
5. When the sensor released, you will get the response of `/homeSw`, and the swState will be 0.
6. [`/hardStop`](https://ponoor.com/en/docs/step-series/osc-command-reference/motor-control/#hardstop_intmotorid) to stop the motor immediately.
7. [`/resetPos`](https://ponoor.com/en/docs/step-series/osc-command-reference/position-management/#resetpos_intmotorid) to reset the current position to 0.

This is bit complicated way but you can try without any firmware update.
Sorry for inconvenience and I will work for updating the firmware anyway!

kanta

**Comment by halpb** [comment]
Kanta-

Thanks for the quick reply!  I just worked through yesterday on how to update the firmware of the boards, so that isn't an issue.

If you move forward with a command for setting the min speed, it would also be nice to have that listed in the configuration tool so users can load that on boot via SD Card.

I attempted to work through the individual commands, but missed a few steps (that you have more clearly outlined).  I will certainly add this into my Max program now and then patiently wait for a firmware/configuration tool update from you.

Thanks again for making such a great product!

-Brett

**Reply by kanta** [reply]
Hi Brett,

Yes, `minSpeed` should be configurable from the SD card slot. I have started to work on updating the firmware, but it will take some more time as I found many sections which need to be updated. Please wait a little longer!

**Comment by halpb** [comment]
No problem at all, thanks for the hard work!  The work-around works great, just need to add some logic so that it knows how to handle homing from a state in which the HOME_SW is already engaged (negative position).

**Reply by kanta** [reply]
Oh right, that's important to build a stable system! You can use [`/getHomeSw`](https://ponoor.com/en/docs/step-series/osc-command-reference/home-and-limit-sensers/#gethomesw_intmotorid) to get the current state of the home sw.

**Reply by kanta** [reply]
The updated firmware with `/setMinSpeed` is now available from [the repository](https://github.com/ponoor/STEP400/). The config tool is also updated. Documentations are not updated yet but the command is quite simple, you can specify the speed with `/setMinSpeed [motorID] [speed(step/s)]` syntax. 
I will update the documents after the STEP800 firmware update is done.

**Reply by halpb** [reply]
Fantastic!  I really appreciate the effort and support.
